<chrono>: Including header pulls in a dynamic initializer, incompatible with /kernel

**Describe the bug**
With Visual Studio 16.10 (Preview 3), just including `<chrono>` now pulls in a dynamic initializer for ` std::numpunct<char>::id` and ` std::numpunct<wchar_t>::id`, which was previously not the case in 16.9. Presumably this is the result of #1821 or #1870.
Since we use `<chrono>` as part of kernel mode driver projects, this now results in a linker warning:
```
repro.obj : warning LNK4210: .CRT section exists; there may be unhandled static initializers or terminators
```
Since the kernel mode build environment does not have startup code that calls these initializers, this is problematic. For these specific initializers the warning is benign in the sense formatting-related chrono functionality cannot be consumed anyway. However, suppressing the warning globally (in our project) is also undesirable because it would allow developers to sneak in their own initializers which are never invoked.

It's clear to us that `<chrono>` isn't officially supported for consumption in the kernel mode build environment and isn't really a part of the freestanding subset. However, since this is a regression, it seems appropriate to bring it to your attention for consideration.

**Command-line test case**
```
C:\...\repro>type repro.cpp
#include <chrono>

extern "C" int atexit(void (__cdecl *func)(void));

int atexit(void (__cdecl *func)(void))
{
  return 0;
}

C:\...\repro>cl /EHsc /Z7 /std:c++latest /permissive- /GS- /LDd repro.cpp /link /noentry /nodefaultlib
Microsoft (R) Incremental Linker Version 14.29.30035.0
Copyright (C) Microsoft Corporation.  All rights reserved.
...
repro.obj : warning LNK4210: .CRT section exists; there may be unhandled static initializers or terminators
```
while with 16.9:
```
C:\...\repro>cl /EHsc /Z7 /std:c++latest /permissive- /GS- /LDd repro.cpp /link /noentry /nodefaultlib
Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29915 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
...
repro.cpp
...
repro.obj
```
with no warning.

**Expected behavior**
Ideally just including `<chrono>` without consuming any of the new formatting related functionality won't pull in any redundant dynamic initializers.

**STL version**
```
Microsoft Visual Studio Community 2019 Preview
Version 16.10.0 Preview 3.0
```


**Additional context**
The repro command lines above do not use `/kernel` or the kernel mode build environment, but simulate the same situation by not linking with the usual user mode startup code.

--- a/stl/inc/format
+++ b/stl/inc/format
@@ -3420,7 +3420,7 @@ _NODISCARD auto make_wformat_args(_Args&&... _Vals) {
 _EXPORT_STD template <output_iterator<const char&> _OutputIt>
 _OutputIt vformat_to(_OutputIt _Out, const string_view _Fmt, const format_args _Args) {
     // Make `_Parse_format_string` type-dependent to defer instantiation:
-    using _Dependent_char = conditional_t<is_same_v<_OutputIt, _Fmt_it>, char, char>;
+    using _Dependent_char = decltype((void) _Out, char{});
     if constexpr (is_same_v<_OutputIt, _Fmt_it>) {
         _Format_handler<_Dependent_char> _Handler(_Out, _Fmt, _Args);
         _Parse_format_string(_Fmt, _Handler);
@@ -3436,7 +3436,7 @@ _OutputIt vformat_to(_OutputIt _Out, const string_view _Fmt, const format_args _
 _EXPORT_STD template <output_iterator<const wchar_t&> _OutputIt>
 _OutputIt vformat_to(_OutputIt _Out, const wstring_view _Fmt, const wformat_args _Args) {
     // Make `_Parse_format_string` type-dependent to defer instantiation:
-    using _Dependent_char = conditional_t<is_same_v<_OutputIt, _Fmt_wit>, wchar_t, wchar_t>;
+    using _Dependent_char = decltype((void) _Out, wchar_t{});
     if constexpr (is_same_v<_OutputIt, _Fmt_wit>) {
         _Format_handler<_Dependent_char> _Handler(_Out, _Fmt, _Args);
         _Parse_format_string(_Fmt, _Handler);
@@ -3452,7 +3452,7 @@ _OutputIt vformat_to(_OutputIt _Out, const wstring_view _Fmt, const wformat_args
 _EXPORT_STD template <output_iterator<const char&> _OutputIt>
 _OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const string_view _Fmt, const format_args _Args) {
     // Make `_Parse_format_string` type-dependent to defer instantiation:
-    using _Dependent_char = conditional_t<is_same_v<_OutputIt, _Fmt_it>, char, char>;
+    using _Dependent_char = decltype((void) _Out, char{});
     if constexpr (is_same_v<_OutputIt, _Fmt_it>) {
         _Format_handler<_Dependent_char> _Handler(_Out, _Fmt, _Args, _Lazy_locale{_Loc});
         _Parse_format_string(_Fmt, _Handler);
@@ -3468,7 +3468,7 @@ _OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const string_view _Fmt,
 _EXPORT_STD template <output_iterator<const wchar_t&> _OutputIt>
 _OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const wstring_view _Fmt, const wformat_args _Args) {
     // Make `_Parse_format_string` type-dependent to defer instantiation:
-    using _Dependent_char = conditional_t<is_same_v<_OutputIt, _Fmt_wit>, wchar_t, wchar_t>;
+    using _Dependent_char = decltype((void) _Out, wchar_t{});
     if constexpr (is_same_v<_OutputIt, _Fmt_wit>) {
         _Format_handler<_Dependent_char> _Handler(_Out, _Fmt, _Args, _Lazy_locale{_Loc});
         _Parse_format_string(_Fmt, _Handler);

--- a/stl/inc/format
+++ b/stl/inc/format
@@ -3417,8 +3417,10 @@ _NODISCARD auto make_wformat_args(_Args&&... _Vals) {
     return _Format_arg_store<wformat_context, _Args...>{_Vals...};
 }
 
-_EXPORT_STD template <output_iterator<const char&> _OutputIt>
-_OutputIt vformat_to(_OutputIt _Out, const string_view _Fmt, const format_args _Args) {
+_EXPORT_STD template <output_iterator<const char&> _OutputIt,
+    class _Sv = string_view> // lazy instantiation, see GH-1926
+_OutputIt vformat_to(_OutputIt _Out, const type_identity_t<_Sv> _Fmt, const format_args _Args) {
+    static_assert(is_same_v<_Sv, string_view>, "misspecified template argument");
     if constexpr (is_same_v<_OutputIt, _Fmt_it>) {
         _Format_handler<char> _Handler(_Out, _Fmt, _Args);
         _Parse_format_string(_Fmt, _Handler);
@@ -3431,8 +3433,10 @@ _OutputIt vformat_to(_OutputIt _Out, const string_view _Fmt, const format_args _
     }
 }
 
-_EXPORT_STD template <output_iterator<const wchar_t&> _OutputIt>
-_OutputIt vformat_to(_OutputIt _Out, const wstring_view _Fmt, const wformat_args _Args) {
+_EXPORT_STD template <output_iterator<const wchar_t&> _OutputIt,
+    class _Sv = wstring_view> // lazy instantiation, see GH-1926
+_OutputIt vformat_to(_OutputIt _Out, const type_identity_t<_Sv> _Fmt, const wformat_args _Args) {
+    static_assert(is_same_v<_Sv, wstring_view>, "misspecified template argument");
     if constexpr (is_same_v<_OutputIt, _Fmt_wit>) {
         _Format_handler<wchar_t> _Handler(_Out, _Fmt, _Args);
         _Parse_format_string(_Fmt, _Handler);
@@ -3445,8 +3449,10 @@ _OutputIt vformat_to(_OutputIt _Out, const wstring_view _Fmt, const wformat_args
     }
 }
 
-_EXPORT_STD template <output_iterator<const char&> _OutputIt>
-_OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const string_view _Fmt, const format_args _Args) {
+_EXPORT_STD template <output_iterator<const char&> _OutputIt,
+    class _Sv = string_view> // lazy instantiation, see GH-1926
+_OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const type_identity_t<_Sv> _Fmt, const format_args _Args) {
+    static_assert(is_same_v<_Sv, string_view>, "misspecified template argument");
     if constexpr (is_same_v<_OutputIt, _Fmt_it>) {
         _Format_handler<char> _Handler(_Out, _Fmt, _Args, _Lazy_locale{_Loc});
         _Parse_format_string(_Fmt, _Handler);
@@ -3459,8 +3465,10 @@ _OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const string_view _Fmt,
     }
 }
 
-_EXPORT_STD template <output_iterator<const wchar_t&> _OutputIt>
-_OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const wstring_view _Fmt, const wformat_args _Args) {
+_EXPORT_STD template <output_iterator<const wchar_t&> _OutputIt,
+    class _Sv = wstring_view> // lazy instantiation, see GH-1926
+_OutputIt vformat_to(_OutputIt _Out, const locale& _Loc, const type_identity_t<_Sv> _Fmt, const wformat_args _Args) {
+    static_assert(is_same_v<_Sv, wstring_view>, "misspecified template argument");
     if constexpr (is_same_v<_OutputIt, _Fmt_wit>) {
         _Format_handler<wchar_t> _Handler(_Out, _Fmt, _Args, _Lazy_locale{_Loc});
         _Parse_format_string(_Fmt, _Handler);

